public/app/features/explore/Explore.tsx

```diff
@@ -347,7 +347,16 @@ export class Explore extends React.PureComponent<ExploreProps, ExploreState> {
                 }
 
                 return (
-                  <main className={cx('m-t-2', styles.logsMain, styles.fullHeight)} style={{ width }}>
+                  <main
+                    className={cx(
+                      'm-t-2',
+                      styles.logsMain,
+                      // We need height to be 100% for tracing iframe to look good but in case of metrics mode
+                      // it makes graph and table also full page high when they do not need to be.
+                      mode === ExploreMode.Tracing && styles.fullHeight
```

Comment:
Did not know you can do use expressions like this for conditional claseses :)  I have only seen `cx(styles.logsMain, { [styles.fullHeight]: mode === ExploreMode.Tracing})`

Reply:
cx should just filter any arg that is falsy. Seems like easier to read in case of single condition like this, in case of multiple I opt for the object but no strong feelings about that.

Reply:
agree